packetbeat/protos/sip/plugin.go

```diff
@@ -67,64 +63,94 @@ func New(
 ) (protos.Plugin, error) {
 	cfgwarn.Beta("packetbeat SIP protocol is used")
 
-	p := &plugin{}
+	isDebug = logp.IsDebug("sip")
+	isDetailed = logp.IsDebug("sipdetailed")
+
 	config := defaultConfig
 	if !testMode {
 		if err := cfg.Unpack(&config); err != nil {
 			return nil, err
 		}
 	}
 
-	if err := p.init(results, watcher, &config); err != nil {
-		return nil, err
+	tp := strings.ToLower(config.TransportProto)
+	switch tp {
+	case "tcp", "udp":
+		p := &plugin{}
+		p.init(results, watcher, &config)
+		return p, nil
 	}
-	return p, nil
+
+	return nil, fmt.Errorf("unsupported transport_protocol: %s", tp)
 }
 
 // Init initializes the HTTP protocol analyser.
-func (p *plugin) init(results protos.Reporter, watcher procs.ProcessesWatcher, config *config) error {
-	p.setFromConfig(config)
-
-	isDebug = logp.IsDebug("sip")
-	isDetailed = logp.IsDebug("sipdetailed")
+func (p *plugin) init(results protos.Reporter, watcher procs.ProcessesWatcher, config *config) {
 	p.results = results
 	p.watcher = watcher
-	return nil
-}
-
-func (p *plugin) setFromConfig(config *config) {
-	p.ports = config.Ports
-	p.keepOriginal = config.KeepOriginal
-	p.parseAuthorization = config.ParseAuthorization
-	p.parseBody = config.ParseBody
+	p.cfg = config
 }
 
 func (p *plugin) GetPorts() []int {
-	return p.ports
+	return p.cfg.Ports
 }
 
 func (p *plugin) ParseUDP(pkt *protos.Packet) {
 	defer logp.Recover("SIP ParseUDP exception")
 
 	if err := p.doParse(pkt); err != nil {
-		debugf("error: %s", err)
+		logp.Error(err)
 	}
 }
 
+// Parse function is used to process TCP payloads.
+func (p *plugin) Parse(
+	pkt *protos.Packet,
+	tcptuple *common.TCPTuple,
+	dir uint8,
+	private protos.ProtocolData,
+) protos.ProtocolData {
+	defer logp.Recover("SIP Parse exception")
+	if err := p.doParse(pkt); err != nil {
```

Comment:
I think we're going to need some stream buffering mechanism in case the SIP request/response is split into multiple TCP segments.